core/templates/dev/head/profile/Preferences.js

```diff
@@ -35,6 +36,46 @@ oppia.controller('Preferences', ['$scope', '$http', '$rootScope', '$modal', '$ti
     _saveDataItem('user_bio', userBio);
   };
 
+  $scope.changedAtLeastOnce = false;
+  $scope.subjectInterestsWarningText = null;
+  $scope.TAG_REGEX = '^[a-z ]+$';
+
+  $scope.updateSubjectInterestsWarning = function(subjectInterests) {
+    var alpha = /^[A-Za-z]+$/;
```

Comment:
OK, so -- you have a string and you want to convert it into a regex, so that both represent the same regular expression. You need to pass the string into the directive, and you need to use the regex in the JS.

What you are currently doing is passing the regex into the directive...